pkg/controller/workspace/tfc_org.go

```diff
@@ -103,6 +103,28 @@ func (t *TerraformCloudClient) SetTerraformVersion(workspace, terraformVersion s
 	return nil
 }
 
+func (t *TerraformCloudClient) updateAgentPoolID(instance *appv1alpha1.Workspace, workspace *tfc.Workspace) error {
+	if instance.Spec.AgentPoolID == workspace.AgentPoolID {
+		return nil
+	}
+
+	updateOptions := tfc.WorkspaceUpdateOptions{
+		AgentPoolID: &instance.Spec.AgentPoolID,
+	}
+
+	if instance.Spec.AgentPoolID != "" {
+		updateOptions.ExecutionMode = tfc.String("agent")
+	} else {
+		updateOptions.ExecutionMode = tfc.String("")
+	}
+
+	_, err := t.Client.Workspaces.Update(context.Background(), t.Organization, workspace.Name, updateOptions)
```

Comment:
Since we haven't implemented a proper context yet, it is best practice to put a `context.TODO()` here. Then later, we can replace them all with real contexts. https://golang.org/pkg/context/#TODO

```suggestion
	_, err := t.Client.Workspaces.Update(context.TODO(), t.Organization, workspace.Name, updateOptions)
```